sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/join/CoGroupByKeyTest.java

```diff
@@ -424,19 +428,31 @@ public void testConsumingDoFn() throws Exception {
             .and(addressesTag, Arrays.asList("8a"))
             .and(namesTag, new ArrayList<>());
 
-    List<KV<String, Integer>> results =
-        DoFnTester.of(
-            new CorrelatePurchaseCountForAddressesWithoutNamesFn(
-                purchasesTag,
-                addressesTag,
-                namesTag))
-                .processBundle(
-                    KV.of(1, result1),
-                    KV.of(2, result2),
-                    KV.of(3, result3),
-                    KV.of(4, result4));
-
-    assertThat(results, containsInAnyOrder(KV.of("4a", 2), KV.of("8a", 0)));
+    KvCoder<Integer, CoGbkResult> coder = KvCoder.of(
+        VarIntCoder.of(),
+        CoGbkResult.CoGbkResultCoder.of(
+            CoGbkResultSchema.of(
+                ImmutableList.of(purchasesTag, addressesTag, namesTag)),
+            UnionCoder.of(
+                ImmutableList.of(
+                    StringUtf8Coder.of(),
+                    StringUtf8Coder.of(),
+                    StringUtf8Coder.of()))));
+
+    PCollection<KV<String, Integer>> results =
+        p.apply(
+                TestStream.create(coder)
```

Comment:
Hm why not just `Create.of()` here too?

Reply:
Fair point. I was being pretty mechanical in my conversion. Done.